src/lib/mina_compile_config/mina_compile_config.ml

```diff
@@ -4,45 +4,95 @@
     tests and production.
 *)
 
-let curve_size = Node_config.curve_size
-
-let default_transaction_fee_string = Node_config.default_transaction_fee
-
-let default_snark_worker_fee_string = Node_config.default_snark_worker_fee
-
-let minimum_user_command_fee_string = Node_config.minimum_user_command_fee
-
-let itn_features = Node_config.itn_features
-
-let compaction_interval_ms = Node_config.compaction_interval
-
-let block_window_duration_ms = Node_config.block_window_duration
-
-let vrf_poll_interval_ms = Node_config.vrf_poll_interval
-
-let rpc_handshake_timeout_sec = Node_config.rpc_handshake_timeout_sec
-
-let rpc_heartbeat_timeout_sec = Node_config.rpc_heartbeat_timeout_sec
-
-let rpc_heartbeat_send_every_sec = Node_config.rpc_heartbeat_send_every_sec
-
-let zkapp_proof_update_cost = Node_config.zkapp_proof_update_cost
-
-let zkapp_signed_pair_update_cost = Node_config.zkapp_signed_pair_update_cost
-
-let zkapp_signed_single_update_cost =
-  Node_config.zkapp_signed_single_update_cost
-
-let zkapp_transaction_cost_limit = Node_config.zkapp_transaction_cost_limit
-
-let max_event_elements = Node_config.max_event_elements
-
-let max_action_elements = Node_config.max_action_elements
-
-let network_id = Node_config.network
-
-let zkapp_cmd_limit = Node_config.zkapp_cmd_limit
-
-let zkapp_cmd_limit_hardcap = Node_config.zkapp_cmd_limit_hardcap
-
-let zkapps_disabled = Node_config.zkapps_disabled
+open Core_kernel
+
+type t =
+  { curve_size : int
+  ; default_transaction_fee_string : string
+  ; default_snark_worker_fee_string : string
+  ; minimum_user_command_fee_string : string
+  ; itn_features : bool
+  ; compaction_interval_ms : int option
+  ; block_window_duration_ms : int
+  ; vrf_poll_interval_ms : int
+  ; network_id : string
+  ; zkapp_cmd_limit : int option
+  ; rpc_handshake_timeout_sec : float
+  ; rpc_heartbeat_timeout_sec : float
+  ; rpc_heartbeat_send_every_sec : float
+  ; zkapp_proof_update_cost : float
+  ; zkapp_signed_pair_update_cost : float
+  ; zkapp_signed_single_update_cost : float
+  ; zkapp_transaction_cost_limit : float
+  ; max_event_elements : int
+  ; max_action_elements : int
+  ; zkapp_cmd_limit_hardcap : int
+  ; zkapps_disabled : bool
+  }
+[@@deriving bin_io_unversioned, to_yojson]
+
+(*TODO: Delete this module and read in a value from the environment*)
+module Compiled = struct
```

Comment:
The comment states as much but I want to be clear, this is  a temporary value which will be deleted when we are constructing this from the environment